### PR TITLE
[ci, model] test: add v5 loader-path test and glm_moe_dsa coverage

### PR DESCRIPTION
**Pull request by TimYangst**
### What does this PR do?

> Closes two gaps in `tests/models/test_models_logits_equal_v5.py`
> compared to its v4 sibling. Tests-only — no production code changes.
>
> 1. **The disk-backed loader path was not tested.** The existing v5
>    test loads weights via in-memory `model.load_state_dict(...)`,
>    which skips `init_empty_weights()`, the safetensors round-trip,
>    and non-persistent buffer rematerialisation — i.e. the path real
>    users actually hit. v4 already had a sibling test for this.
> 2. **`glm_moe_dsa` had no coverage.** The model has shipped patchgen
>    output for a while but had no toy config and no parametrised case.
>
> Follow-up to #722 (v5 bitwise tests).

### Checklist Before Starting

- Related: #721 (v4 bitwise tests), #722 (v5 bitwise tests, the scope
  this PR extends).
- PR title `[ci, model]` matches the precedent set by #721 / #722 for
  test additions that exercise specific model classes.

### Test

> All 26 cases pass locally on a single H100 (was 18 before this PR):
>
> ```
> tests/models/test_models_logits_equal_v5.py ..........................   [100%]
> ======================== 26 passed in 65.72s (0:01:05) =========================
> ```
>
> New cases:
> - `glm_moe_dsa-{eager,sdpa}` — eager/fp32 baseline + sdpa/bf16 (the
>   close-to-real-user path). FA2 is intentionally skipped: upstream
>   sets `_supports_flash_attn = False`.
> - 6 loader cases (`_LOADER_CASES`) covering every v5 MoE family with
>   at least one representative attn impl: `qwen3_moe-{eager,fa2}`,
>   `glm_moe_dsa-{eager,sdpa}`, `qwen3_vl_moe-fa2`, `qwen3_omni_moe-fa2`.
>
> `make quality` and pre-commit hooks pass.

### API and Usage Example

> N/A — tests-only.

### Design & Code Changes

> **1. Module-level transformers gate bumped from `>= 5.0.0` to `>= 5.2.0`.**
> The v5 environment in this repo (`transformers5-exp` extra in
> `pyproject.toml`) is pinned to 5.2.0, and `glm_moe_dsa` strictly
> needs 5.2.0. A single tighter gate is simpler than per-case
> `min_transformers_version` overrides.
>
> **2. New toy config `tests/toy_config/glm_moe_dsa_toy/`.**
> Derived from `GlmMoeDsaConfig` defaults: shrunk to a 4-layer,
> 256-hidden, 8-routed-expert toy. `num_hidden_layers=4` keeps
> `mlp_layer_types` covering both `dense` and `sparse` (default split
> is 3 + rest); `index_topk=16` stays ≤ the test fixture's `seq_len=32`.
> README documents every delta from upstream.
>
> **3. Two `glm_moe_dsa` cases in `CASES`** — `eager+fp32` (RNG
> baseline) and `sdpa+bf16`. Both pass `_experts_implementation="eager"`
> so HF stays on the eager expert loop instead of `torch._grouped_mm`,
> matching the qwen3_moe convention.
>
> **4. New `test_logits_bitwise_equal_v5_via_loader`** parametrised
> over `_LOADER_CASES`. Writes the HF state-dict to a tmp dir, rebuilds
> via `build_foundation_model(weights_path=<dir>)`, then bitwise-compares
> logits. Mirrors v4's `test_logits_bitwise_equal_via_runtime_converter`,
> including the non-persistent buffer (rotary `inv_freq`) restoration
> workaround for the `init_empty_weights() + register_buffer` ULP drift
> documented in the v4 test's docstring.
>
> Two design choices worth flagging:
>
> - **`config_path` is a directory, not the in-memory config object.**
>   Some VeOmni models register a `MODEL_CONFIG_REGISTRY` hook that
>   mutates the config at build time (e.g. `qwen3_omni_moe` forces
>   `tie_word_embeddings=False` so the loader's post-load tie is a
>   no-op on the thinker+talker container). That hook only fires when
>   `config_path` is a string path — handing in a `PretrainedConfig`
>   short-circuits it. We re-apply `case.config_overrides` via
>   `config_kwargs` so private fields like `_experts_implementation`
>   survive without depending on JSON round-trip semantics for
>   underscored attributes.
> - **The buffer-restoration loop iterates the VeOmni model's buffers
>   and looks them up on the HF side, not the other direction.** The
>   VeOmni model can be a *subset* of the HF model (`qwen3_omni_moe`
>   strips `talker` because it isn't subclassed locally), so HF carries
>   buffers with no VeOmni counterpart — direction matters.
>
> **Scope choices documented inline:**
> - The loader test omits `qwen3_5_moe`: that case extracts
>   `full_config.text_config` and switches `model_type` from
>   `qwen3_5_moe` to `qwen3_5_moe_text`. VeOmni's `MODELING_REGISTRY`
>   keys off the parent `model_type`, so handing the extracted
>   text-config to `build_foundation_model(config_path=...)` would
>   need either a synthetic wrapper config or a registry-bypass — both
>   would misrepresent the real-user loader flow.
> - For v5 MoE, on-disk layout already matches VeOmni's stacked layout,
>   so no expert-stacking converter fires here. The value of this test
>   is exercising the loader pipeline (`init_empty_weights`, safetensors
>   deserialisation, buffer rematerialisation), not the converter.
> - `qwen3_5` / `qwen3_5_moe` stay text-only for now (no `vlm_full`
>   case). Multimodal coverage is blocked on a VeOmni patchgen bug:
>   patched `Qwen3_5Model.forward` calls
>   `compute_3d_position_ids(..., mm_token_type_ids=...)` but doesn't
>   override `compute_3d_position_ids` to accept that kwarg, so the
>   VeOmni multimodal path raises `TypeError`. Tracked via an inline
>   TODO above the qwen3_5 text-only cases; will be addressed in a
>   follow-up that fixes the four patchgen configs.

### Checklist Before Submitting

- [x] Read the [Contribute Guide](https://github.com/ByteDance-Seed/VeOmni/blob/main/CONTRIBUTING.md)
- [x] Applied pre-commit checks
- [x] Added/updated documentation (README in the new toy_config dir)
- [x] Added tests to CI workflow (this PR *is* the test addition; reuses
      the existing v5-tests CI job from #725)

### CODE REVIEW

File: tests/models/test_models_logits_equal_v5.py [hunk 1]
```diff
@@ -499,3 +532,207 @@ def test_logits_bitwise_equal_v5(case: Case):
...
+    persistent_keys = set(hf_state_dict.keys())
+    for name, buf in hf_buffers.items():
+        if name in persistent_keys:
+            continue
+        parts = name.split(".")
+        target_module = model
+        for p in parts[:-1]:
+            target_module = getattr(target_module, p)
+        target = target_module._buffers[parts[-1]]
+        target.copy_(buf.to(target.device, dtype=target.dtype))
```
[the remark this answers is not in the document]
Comment (TimYangst) [reply tests/models/test_models_logits_equal_v5.py]: getattr(module_list, "0") works fine — nn.Module.__getattr__ looks up _modules, where ModuleList stores children under string keys. The same pattern runs in the v4 sibling test against multi-layer models in CI without issue, and all 4 loader cases here pass. get_buffer is cleaner but I'd rather refactor v4 + v5 together in a separate PR.

File: tests/models/test_models_logits_equal_v5.py [hunk 2]
```diff
@@ -499,3 +532,207 @@ def test_logits_bitwise_equal_v5(case: Case):
...
+
+    model_hf = _build_hf_model(case, config, target_dtype)
+    with torch.no_grad():
+        logits_hf = (
+            _forward_target(model_hf, case)(input_ids=input_ids.clone(), use_cache=False, **fwd_kwargs)
+            .logits.detach()
+            .clone()
+        )
+    hf_state_dict = copy.deepcopy(model_hf.state_dict())
+    hf_buffers = {n: b.detach().clone() for n, b in model_hf.named_buffers()}
```
[the remark this answers is not in the document]
Comment (TimYangst) [reply tests/models/test_models_logits_equal_v5.py:737]: oy configs only (4 layers, hidden=256) — state_dict is tens of MB, no OOM risk. Keeping copy.deepcopy here for parity with the v4 sibling test, which uses the same pattern.

**Review by Luosuu — APPROVED** [review]
